app/src/main/java/com/example/weatherapp/MainActivity.kt

```diff
@@ -1,11 +1,93 @@
+import android.os.Build
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.os.IBinder
+import androidx.core.content.ContextCompat
+import com.example.weatherapp.databinding.ActivityMainBinding
+import com.example.weatherapp.services.LocalizationService
 
 class MainActivity : AppCompatActivity() {
+
+    private lateinit var broadcastReceiver: BroadcastReceiver
+    private lateinit var mService: LocalizationService
+    private lateinit var binding: ActivityMainBinding
+    private var mBound: Boolean = false
+
+    private val connection = object : ServiceConnection {
+        override fun onServiceConnected(name: ComponentName, service: IBinder) {
+            val binder = service as LocalizationService.ServiceBinder
+            mService = binder.getService()
+            mBound = true
+        }
+
+        override fun onServiceDisconnected(componentName: ComponentName) {
+            mBound = false
+        }
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        if(!checkPermissions())
+            enableButton()
+    }
+
+    override fun onResume() {
+        super.onResume()
+
+        broadcastReceiver = object : BroadcastReceiver() {
+            override fun onReceive(context: Context?, intent: Intent) {
+                binding.textViewLocalization.text = intent.getStringExtra("cityName")
+            }
+        }
+        registerReceiver(broadcastReceiver, IntentFilter("locationUpdate"))
+    }
+
+    override fun onStop() {
+        super.onStop()
+        unbindService(connection)
+        mBound = false
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        unregisterReceiver(broadcastReceiver)
+    }
+
+    fun enableButton() {
+        binding.buttonGetWeather.setOnClickListener {
+            Intent(this, LocalizationService::class.java).also { intent ->
+                bindService(intent, connection, Context.BIND_AUTO_CREATE)
+            }
+        }
+    }
+
+    private fun checkPermissions() : Boolean {
+        if (Build.VERSION.SDK_INT >= 23 && ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION)
+            != PackageManager.PERMISSION_GRANTED && ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION)
+            != PackageManager.PERMISSION_GRANTED) {
+            requestPermissions(arrayOf(Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION), 100)
+            return true
+        }
+        return false
+    }
+
+    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+        if(requestCode == 100){
```

Comment:
please add comment  in code about what "100" means

Reply:
done